lib/daru/io/exporters/json.rb

```diff
@@ -0,0 +1,250 @@
+        #   #   {
+        #   #     "common": {
+        #   #       "gender": "Male",
+        #   #       "age": 18
+        #   #     },
+        #   #     "specific": {
+        #   #       "name": "Jon Snow"
+        #   #     }
+        #   #   },
+        #   #   {
+        #   #     "common": {
+        #   #       "gender": "Male",
+        #   #       "age": 54
+        #   #     },
+        #   #     "specific": {
+        #   #       "name": "Rhaegar Targaryen"
+        #   #     }
+        #   #   },
+        #   #   {
+        #   #     "common": {
+        #   #       "gender": "Female",
+        #   #       "age": 36
+        #   #     },
+        #   #    "specific": {
+        #   #       "name": "Lyanna Stark"
+        #   #     }
+        #   #   }
+        #   # ]
+        #
+        # @example Writing to a JSON file with dynamic JsonPaths
+        #   df = Daru::DataFrame.new(
+        #     [
+        #       {name: 'Jon Snow', age: 18, sex: 'Male'},
+        #       {name: 'Rhaegar Targaryen', age: 54, sex: 'Male'},
+        #       {name: 'Lyanna Stark', age: 36, sex: 'Female'}
+        #     ],
+        #     order: %i[name age sex],
+        #     index: %i[child dad mom]
+        #   )
+        #
+        #   #=> #<Daru::DataFrame(3x3)>
+        #   #            name          age       sex
+        #   # child   Jon Snow         18       Male
+        #   # dad   Rhaegar Ta         54       Male
+        #   # mom   Lyanna Sta         36     Female
+        #
+        #   Daru::IO::Exporters::JSON.new(
+        #     df,
+        #     'filename.json',
+        #     orient: :records,
+        #     pretty: true,
+        #     age: '$.{name}.age',
+        #     sex: '$.{name}.gender'
+        #   )
+        #
+        #   #=>
+        #   # [
+        #   #   {
+        #   #     "Jon Snow": {
+        #   #       "gender": "Male",
+        #   #       "age": 18
+        #   #     }
+        #   #   },
+        #   #   {
+        #   #     "Rhaegar Targaryen": {
+        #   #       "gender": "Male",
+        #   #       "age": 54
+        #   #     }
+        #   #   },
+        #   #   {
+        #   #     "Lyanna Stark": {
+        #   #     "gender": "Female",
+        #   #     "age": 36
+        #   #     }
+        #   #   }
+        #   # ]
+        def initialize(dataframe, path, pretty: false, **jsonpaths)
+          require 'json'
+          optional_gem 'jsonpath'
+
+          super(dataframe)
+          @path          = path
+          @pretty        = pretty
+          @jsonpath_hash = jsonpaths.empty? ? nil : jsonpaths
+        end
+
+        def call
+          @jsonpath_hash ||= @dataframe.vectors.to_a.map { |v| {v => "$.#{v}"} }.reduce(:merge)
+          @vectors         = @jsonpath_hash.keys
+          @jsonpaths       = process_jsonpath
+          @json_content    = @dataframe.map_rows_with_index do |row, idx|
+            init_hash(@jsonpaths, @vectors, row, idx)
+          end
+
```

Comment:
As I've written in private, let's create GH issue with 5-7 examples of reasonable target outputs of the same dataframe, and brainstorm the minimal flexible API?..